### ISSUE TITLE
Containers utilizing multiple networks, with a restart policy, do not reattach to all networks on container host reboot.

### ISSUE THREAD

**Issue by robdaemon** [docker/docker#17526]
Description of problem:

Containers utilizing multiple networks, with a restart policy, do not reattach to all networks on container host reboot.

`docker version`:

Client:
 Version:      1.9.0-rc3
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   2100b94
 Built:        Wed Oct 28 01:29:10 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.0-rc3
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   2100b94
 Built:        Wed Oct 28 01:29:10 UTC 2015
 OS/Arch:      linux/amd64

`docker info`:

Containers: 15
Images: 124
Server Version: 1.9.0-rc3
Storage Driver: devicemapper
 Pool Name: docker-253:33-1048579-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 7.406 GB
 Data Space Total: 107.4 GB
 Data Space Available: 33.04 GB
 Metadata Space Used: 10.71 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.137 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /data/docker/devicemapper/devicemapper/data
 Metadata loop file: /data/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.77 (2012-10-15)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-31-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 4
Total Memory: 7.797 GiB
Name: rob-core
ID: 4CDW:4T54:CBF7:B3B5:NM5Q:7UUL:J7OR:6GNT:ZZ6W:GBAL:V6QQ:NK7V
Labels:
 com.docker.network.driver.overlay.bind_interface=eth0
Cluster store: etcd://10.0.0.249:3379
Cluster advertise: 10.0.0.249:2376

`uname -a`:

Linux rob-core 3.19.0-31-generic #36~14.04.1-Ubuntu SMP Thu Oct 8 10:21:08 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

Environment details (AWS, VirtualBox, physical, etc.):

HP Public Cloud (OpenStack)

How reproducible:

Every time

Steps to Reproduce:
1. Run a container, attached to a specific network with `--net=<network>` and `--restart=always`
2. Attach the container to a second network with `docker network connect <network> <container>`
3. Reboot container host
4. Observe that container is only connected to the `--net=<network>` network, and not both networks.

Actual Results:

Container only reconnects to the network specified in the `--net=<network>` command line parameter.

Expected Results:

Container should be connected to all connected networks at reboot.

Additional info:

I sort of understand why this happens - it is a consequence of not allowing multiple networks to be specified on the `docker run` command line. We need a mechanism to make the network connections persist across container host reboots, or the ability to specify multiple networks on the same `run` command.

**Comment by mrjana** [comment]
@robdaemon It is expected to reconnect to all networks on daemon restart with a `restart=always" policy. If it is not it is a bug. But can you try it after #17514 is merged?

**Comment by robdaemon** [comment]
Awesome - I'll try it again when that gets merged.

**Comment by mavenugo** [comment]
@robdaemon i could reproduce this even with #17514 in place. So, pls hold on till we resolve this issue.

Again thanks for raising this issue.

**Comment by mavenugo** [comment]
@robdaemon RC4 images are available https://github.com/docker/docker/pull/17000#issuecomment-152687350 Pls give it a try.

**Comment by robdaemon** [comment]
Unfortunately this seems to have made the problem much worse. I'll open a new issue.
